internal/formats/github/encoder.go

```diff
@@ -0,0 +1,158 @@
+package github
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/internal/version"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+// toGithubModel converts the provided SBOM to a GitHub dependency model
+func toGithubModel(s *sbom.SBOM) DependencySnapshot {
+	scanTime := time.Now().Format(time.RFC3339) // TODO is there a record of this somewhere?
+	v := version.FromBuild().Version
+	if v == "[not provided]" {
+		v = "0.0.0-dev"
+	}
+	return DependencySnapshot{
+		Version: 0,
+		// TODO allow property input to specify the Job, Sha, and Ref
+		Detector: DetectorMetadata{
+			Name:    internal.ApplicationName,
+			URL:     "https://github.com/anchore/syft",
+			Version: v,
+		},
+		Metadata:  toSnapshotMetadata(s),
+		Manifests: toGithubManifests(s),
+		Scanned:   scanTime,
+	}
+}
+
+// toSnapshotMetadata captures the linux distribution information and other metadata
+func toSnapshotMetadata(s *sbom.SBOM) Metadata {
+	out := Metadata{}
+
+	if s.Artifacts.LinuxDistribution != nil {
+		d := s.Artifacts.LinuxDistribution
+		qualifiers := packageurl.Qualifiers{}
+		if len(d.IDLike) > 0 {
+			qualifiers = append(qualifiers, packageurl.Qualifier{
+				Key:   "like",
+				Value: strings.Join(d.IDLike, ","),
+			})
+		}
+		purl := packageurl.NewPackageURL("generic", "", d.ID, d.VersionID, qualifiers, "")
+		out["syft:distro"] = purl.ToString()
+	}
+
+	return out
+}
+
+func filesystem(p pkg.Package) string {
+	if len(p.Locations) > 0 {
+		return p.Locations[0].FileSystemID
+	}
+	return ""
+}
+
+// toPath Generates a string representation of the package location, optionally including the layer hash
+func toPath(s source.Metadata, p pkg.Package) string {
+	if len(p.Locations) > 0 {
+		location := &p.Locations[0]
+		switch s.Scheme {
+		case source.ImageScheme:
+			in := strings.ReplaceAll(s.ImageMetadata.UserInput, ":/", "//")
+			path := strings.TrimPrefix(location.RealPath, "/")
+			return fmt.Sprintf("%s:/%s", in, path)
+		case source.FileScheme:
+			return fmt.Sprintf("%s:/%s", s.Path, strings.TrimPrefix(location.RealPath, "/"))
```

Comment:
I was just playing with this, and when I scan individual files, I get this kind of output:

```json
"name": "/Users/dan/homebrew/bin/syft:/Users/dan/homebrew/Cellar/syft/0.41.1/bin/syft",
```

What is the motivation for this string format? I'm wondering if it's easier to read to just include the real path or something, WDYT?

Reply:
The motivation for this was like:
```
/some/tar/archive.tar.gz:/some/path/within/it
```
... but maybe that is really not what is happening here, I can definitely change this... but I think this is actually illustrating one annoyance with our different resolvers: image resolvers have a path relative to the image... we should have paths relative to the directory scanned here instead of full paths

Reply:
I like this scheme for archives (like TAR) FWIW!

For other things, I agree that it's a bit awkward.